android/libraries/rib-android/src/main/kotlin/com/uber/rib/core/RibActivity.kt

```diff
@@ -41,30 +38,32 @@ import com.uber.rib.core.lifecycle.ActivityLifecycleEvent.Companion.create
 import com.uber.rib.core.lifecycle.ActivityLifecycleEvent.Companion.createOnCreateEvent
 import io.reactivex.CompletableSource
 import io.reactivex.Observable
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.rx2.asObservable
 
 /** Base implementation for all VIP [android.app.Activity]s. */
 abstract class RibActivity : CoreAppCompatActivity(), ActivityStarter, LifecycleScopeProvider<ActivityLifecycleEvent>, RxActivityEvents {
   private var router: ViewRouter<*, *>? = null
-  private val lifecycleBehaviorRelay = BehaviorRelay.create<ActivityLifecycleEvent>()
-  private val lifecycleRelay: Relay<ActivityLifecycleEvent> = lifecycleBehaviorRelay.toSerialized()
-  private val callbacksRelay = PublishRelay.create<ActivityCallbackEvent>().toSerialized()
+  private val lifecycleFlow = MutableSharedFlow<ActivityLifecycleEvent>(1, 0, BufferOverflow.DROP_OLDEST)
+  private val callbacksFlow = MutableSharedFlow<ActivityCallbackEvent>(0, 1, BufferOverflow.DROP_OLDEST)
 
   /** @return an observable of this activity's lifecycle events. */
   override fun lifecycle(): Observable<ActivityLifecycleEvent> {
-    return lifecycleRelay.hide()
+    return lifecycleFlow.asObservable()
```

Comment:
Nit, kind of out of scope of the change: I generally prefer to have these cold streams defined as properties instead of doing a new conversion (new allocation!) at every method call, to reduce the number of allocations. Consider:

```kotlin
val flow = MutableSharedFlow<Int>(1, 0, BufferOverflow.DROP_OLDEST)
fun getObservable() = flow.asObservable()

fun main(args: Array<String>) {
    repeat(100) {
        println(getObservable())
    }
}
```
Translates to 1 new allocation every time method is called:
```
io.reactivex.internal.operators.observable.ObservableCreate@7a30d1e6
io.reactivex.internal.operators.observable.ObservableCreate@5891e32e
io.reactivex.internal.operators.observable.ObservableCreate@cb0ed20
io.reactivex.internal.operators.observable.ObservableCreate@8e24743
io.reactivex.internal.operators.observable.ObservableCreate@74a10858
```

if we allocate it as a property once:

```kotlin
val flow = MutableSharedFlow<Int>(1, 0, BufferOverflow.DROP_OLDEST)
val flowAsObservable = flow.asObservable()
fun getObservable() = flowAsObservable

fun main(args: Array<String>) {
    repeat(100) {
        println(getObservable())
    }
}
```
Only one allocation is done:
```
io.reactivex.internal.operators.observable.ObservableCreate@587d1d39
io.reactivex.internal.operators.observable.ObservableCreate@587d1d39
io.reactivex.internal.operators.observable.ObservableCreate@587d1d39
io.reactivex.internal.operators.observable.ObservableCreate@587d1d39
io.reactivex.internal.operators.observable.ObservableCreate@587d1d39
```

Reply:
Yeah, this is a reasonable suggestion. This might save some overhead on autodispose calling these apis.